client/my-sites/plans-grid/components/actions.tsx

```diff
@@ -131,6 +135,26 @@ const SignupFlowPlanFeatureActionButton = ( {
 		);
 	}
 
+	if (
+		! hasFreeTrialPlan &&
+		isBusinessPlan( planSlug ) &&
+		! isEligibleForTrial &&
+		window.location.pathname.indexOf( 'new-hosted-site' ) !== -1
```

Comment:
I think all of these ought to happen outside of plans-grid components. You should also be using `plansIntent` mechanism for filtering anything on respective flows. Again, happy to answer any clarifying questions.

Reply:
There is also a `planActionOverrides` property/structure that may be relevant for this logic to migrate out.

Reply:
Thanks for your feedback @chriskmnds! I'll adapt it to work without state/selectors.

Reply:
@paulopmt1 Feel free to reach out to me pr Dehan if there are any questions today. Christos is off for the weekend 🙂